File: 1D Array/C/program_1.c
/* Write a program, enter n elements into an array, perform 
linear search*/
#include<stdio.h>
int linear_search(int arr[],int n,int target){
     for(int i=0;i<n;i++){
        if(arr[i]== target){
            return i;
        }
     }
     return -1;
}

int main(){
    int n,target;
    printf("Enter the number of element in the array :");
    scanf("%d",&n);
    int arr[n];
    printf("The element in the array is :\n");
    for(int i=0;i<n;i++){
        printf("Enter the element in the array arr[%d]:",i);
        scanf("%d",&arr[i]);
    }
    printf("Enter the target element :");
    scanf("%d",&target);
    int result =linear_search(arr,n,target);
    if(result!=-1){
        printf("The target %d found at the index %d:\n ",target,result);

    }
    else{
        printf("The target %d not found in the array:",target);
    }

    return 0;
}

File: 1D Array/java/program1.java
import java.util.Scanner;

public class program1 {
    public static int linearSearch(int[] arr, int n, int target) {
        for (int i = 0; i < n; i++) {
            if (arr[i] == target) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements in the array: ");
        int n = scanner.nextInt();
        int[] arr = new int[n];
        System.out.println("The elements in the array are:");
        for (int i = 0; i < n; i++) {
            System.out.print("Enter the element in the array arr[" + i + "]: ");
            arr[i] = scanner.nextInt();
        }
        System.out.print("Enter the target element: ");
        int target = scanner.nextInt();
        int result = linearSearch(arr, n, target);
        if (result != -1) {
            System.out.println("The target " + target + " found at the index " + result);
        } else {
            System.out.println("The target " + target + " not found in the array");
        }
        scanner.close();
    }
}